source/Projectile.cpp

```diff
@@ -35,11 +35,44 @@ namespace {
 
 
 
-Angle Projectile::Inaccuracy(double value)
+Angle Projectile::Inaccuracy(double value, double smoothness)
 {
 	Angle inaccuracy;
 	if(value)
-		inaccuracy = Angle::Random(value) - Angle::Random(value);
+	{
+		if(!smoothness)
+			inaccuracy = Angle::Random(2 * value) - Angle(value);
+		else
+		{
+			double randomFactor = Random::Normal();
+			// Invert smoothness so that higher stat values are associated with greater realized smoothness.
+			// Do it here so that the true value of the stat is retained for any other calculations.
+			smoothness = 1. / smoothness;
+			// Multiplying smoothness by 2.317 mimics legacy behavior with a smoothness stat of 1 (default).
+			smoothness *= 2.317;
+			// Compress values above and below the mean into [0, 1]
+			// where the range pulled is determined by smoothness.
+			randomFactor = (randomFactor + smoothness) / (2 * smoothness);
+			// Retain only the fractional information.  This facilitates realized smoothness
+			// as fractional information is often redundant if not fully compressed.
+			// Might be possible to get away with int32_t here, not sure.
+			randomFactor = randomFactor - static_cast<int64_t>(randomFactor);
+			// Push negative values into the usable range.
+			if(randomFactor < 0)
+				randomFactor++;
+
+			// Use negative smoothness to achieve a spread that concentrates at the endpoints rather than the center.
+			if(smoothness < 0)
+			{
+				if(randomFactor < 0.5)
+					randomFactor += 0.5;
+				else
+					randomFactor -= 0.5;
+			}
```

Comment:
You're converting a number into a sample from a distribution based on that number; that implementation belongs in the Random class, not here.

Reply:
I'll move it over.  Question though: do you think it makes sense to leave smoothness as the raw value and manipulate it every time inaccuracy is calculated?  I can just as easily add the comment regarding legacy behavior in Weapon.cpp when it gets assigned and that should save some execution time.

Reply:
Fixed in 28294111da9cd02bdaa8148cf19256cdd7593d6f.